### PR TITLE
#75 Copied YT code from https://github.com/alxhotel/chromecast-api Se…

### PR DESCRIPTION
**Pull request by raintonr**
Just what it says in the title :)

### CODE REVIEW

File: lib/chromecastDevice.js [hunk 1]
```diff
@@ -907,13 +909,37 @@ module.exports = function (adapter) {
...
-            this._mediaPlayer.playUrlPromise(url2play /*org_url2play, streamType*/)
-                .then(() => {
-                    adapter.setState(`${this.mac}.player.url2play`, {val: org_url2play, ack: true});
+            // Is this a Youtube URL?
+            const youtubeId = Youtube.getYoutubeId(url2play)
+            if (youtubeId) {
+                adapter.log.debug('Trying to play YouTube (' + youtubeId + ') on ' + this._connection.name);
+                const client = this._mediaPlayer._client._client;
+
+                // TODO: do something funky with sessions & re-use?
```
Comment (raintonr) [inline lib/chromecastDevice.js:918]: As  I elude to... think something fishy is going on with repeat attempts to play videos and more work required here.

**Comment by Jey-Cee** [comment]
Hi @raintonr,

while i am now maintaining this repository & the castv2-player i came to your PR.
I think we should split your PR into two parts, the youtube lib and the adapter specific part. 
The lib part seems to fit better in the player module and is usable for other projects they build on top of it.

Can/will you do this?

Greets 
Jey Cee

**Comment by raintonr** [comment]
> The lib part seems to fit better in the player module and is usable for other projects they build on top of it.

Yeah - I would be inclined to agree, but thought this would get us going for now. Besides, I just stole that code from another project, surely it would be for them to do the upstream additions they actually created? :wink:

**Comment by Jey-Cee** [comment]
Have you made some additions to their code we can provide them?
I've looked into the original code and there was no changes since 2 years.

**Comment by raintonr** [comment]
> Have you made some additions to their code we can provide them?

No. I just copied it (including a comment saying where it was copied from).

**Comment by Jey-Cee** [comment]
Ok, so will you split it?
If not it's ok for me and i will do it on my own.

**Comment by Apollon77** [comment]
@raintonr WOuld you please rebase your PR based on latest changes on master ... then we integrate it

**Comment by raintonr** [comment]
> @raintonr WOuld you please rebase your PR based on latest changes on master ... then we integrate it

Isn't @Jey-Cee looking after this?

**Comment by Apollon77** [comment]
No, he is not maintaining the adapter anymore, so I am now

**Comment by GermanBluefox** [comment]
@raintonr  Sorry the adapter was refactored. Could you please resolve the conflicts?

**Comment by raintonr** [comment]
Funnily enough I reinstalled the latest version yesterday, so yes, will do. Can't get to this until next week though.

**Comment by raintonr** [comment]
@GermanBluefox this all looks good here now.

**Pull request merged**